gatk-launch

```diff
@@ -276,9 +307,9 @@ def cacheJarOnGCS(jar, dryRun):
             return jar
 
 
-def runGATK(sparkRunner, suppliedSparkSubmitCommand, dryrun, gatkArgs, sparkArgs):
+def runGATK(sparkRunner, suppliedSparkSubmitCommand, dryrun, gatkArgs, sparkArgs, javaOptions):
     if sparkRunner is None or sparkRunner == "LOCAL":
-        cmd = getLocalGatkRunCommand() + gatkArgs + sparkArgs
+        cmd = getLocalGatkRunCommand(javaOptions) + gatkArgs + sparkArgs
         runCommand(cmd, dryrun)
     elif sparkRunner == "SPARK":
```

Comment:
You'll need to hook `--javaOptions` up to the two Spark modes as well in order to satisfy all of the requirements listed in https://github.com/broadinstitute/gatk/issues/2694#issuecomment-305007868

You can do this by adding the Java options to the Spark properties `spark.driver.extraJavaOptions` and `spark.executor.extraJavaOptions` when `sparkRunner == "SPARK"` or `sparkRunner == "GCS"`. These Spark properties are currently initialized in `DEFAULT_SPARK_ARGS` above -- you'll have to refactor to allow them to be modified.